app/controllers/courses_controller.rb

```diff
@@ -24,6 +24,15 @@ def destroy
     head :no_content
   end
 
+  # get visible returns courses that are taught by an admin or enrolled in by a student
+  def get_visible
+    if current_user.admin
```

Comment:
agreed. we need to think hard about how much we separate student and teacher logic. we absolutely cannot afford to mix them up.